.github/workflows/update-docs.yml

```diff
@@ -0,0 +1,46 @@
+# NOTE: repo needs a secrets PRIMER3_TOKEN that is a
+# GitHub Personal Access Token (Fine Grained) with "Contents" Read Write
+# permission
+name: "Push New Tagged Version Docs Build" # Build on a new version tag
+on:
+  push:
+    branches:
+      - "master"
+    tags:
+      - "[0-9]+.[0-9]+.[0-9]+"
+
+jobs:
+  docs:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3.3.0
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v4.4.0
+      with:
+        python-version: '3.10'
+    - name: Install and run sphinx to build the docs
+      run: |
+        python -m pip install --upgrade pip
+        pip install cython myst-parser sphinx sphinx_rtd_theme sphinx-autodoc-typehints
```

Comment:
If we don't want to introduce `cython` or `venv`, we could minimally drop these requirements into context-specific `*-requirements.txt` files, like `doc-build-requirements.txt` and `pip install -r` them.

Reply:
I'm going to resolve this and add a line item to the effort tracker